NPE when opening saved filter

Ticket 275715
c:geo version: 2024.01.06

NPE when opening saved filter

```
03-14 22:35:13.331 17928 17928 E AndroidRuntime: FATAL EXCEPTION: main
03-14 22:35:13.331 17928 17928 E AndroidRuntime: Process: cgeo.geocaching, PID: 17928
03-14 22:35:13.331 17928 17928 E AndroidRuntime: java.lang.RuntimeException: Unable to resume activity {cgeo.geocaching/cgeo.geocaching.maps.google.v2.GoogleMapActivity}: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=456, result=-1, data=Intent { (has extras) }} to activity {cgeo.geocaching/cgeo.geocaching.maps.google.v2.GoogleMapActivity}: java.lang.NullPointerException: Attempt to invoke interface method 'void cgeo.geocaching.maps.interfaces.PositionAndHistory.repaintRequired()' on a null object reference
03-14 22:35:13.331 17928 17928 E AndroidRuntime: 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4324)
03-14 22:35:13.331 17928 17928 E AndroidRuntime: 	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4356)
03-14 22:35:13.331 17928 17928 E AndroidRuntime: 	at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:52)
03-14 22:35:13.331 17928 17928 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
03-14 22:35:13.331 17928 17928 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
03-14 22:35:13.331 17928 17928 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2065)
03-14 22:35:13.331 17928 17928 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:107)
03-14 22:35:13.331 17928 17928 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:214)
03-14 22:35:13.331 17928 17928 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7631)
03-14 22:35:13.331 17928 17928 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
03-14 22:35:13.331 17928 17928 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
03-14 22:35:13.331 17928 17928 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:964)
03-14 22:35:13.331 17928 17928 E AndroidRuntime: Caused by: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=456, result=-1, data=Intent { (has extras) }} to activity {cgeo.geocaching/cgeo.geocaching.maps.google.v2.GoogleMapActivity}: java.lang.NullPointerException: Attempt to invoke interface method 'void cgeo.geocaching.maps.interfaces.PositionAndHistory.repaintRequired()' on a null object reference
03-14 22:35:13.331 17928 17928 E AndroidRuntime: 	at android.app.ActivityThread.deliverResults(ActivityThread.java:4993)
03-14 22:35:13.331 17928 17928 E AndroidRuntime: 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4311)
03-14 22:35:13.331 17928 17928 E AndroidRuntime: 	... 11 more
03-14 22:35:13.331 17928 17928 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'void cgeo.geocaching.maps.interfaces.PositionAndHistory.repaintRequired()' on a null object reference
03-14 22:35:13.331 17928 17928 E AndroidRuntime: 	at cgeo.geocaching.maps.CGeoMap.refreshMapData(CGeoMap.java:842)
03-14 22:35:13.331 17928 17928 E AndroidRuntime: 	at cgeo.geocaching.maps.google.v2.GoogleMapActivity.onActivityResult(GoogleMapActivity.java:281)
03-14 22:35:13.331 17928 17928 E AndroidRuntime: 	at android.app.Activity.dispatchActivityResult(Activity.java:8145)
03-14 22:35:13.331 17928 17928 E AndroidRuntime: 	at android.app.ActivityThread.deliverResults(ActivityThread.java:4986)
03-14 22:35:13.331 17928 17928 E AndroidRuntime: 	... 12 more
03-14 22:35:13.334 17928 17928 W cgeo    : [main] UncaughtException
03-14 22:35:13.334 17928 17928 W cgeo    : java.lang.RuntimeException: Unable to resume activity {cgeo.geocaching/cgeo.geocaching.maps.google.v2.GoogleMapActivity}: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=456, result=-1, data=Intent { (has extras) }} to activity {cgeo.geocaching/cgeo.geocaching.maps.google.v2.GoogleMapActivity}: java.lang.NullPointerException: Attempt to invoke interface method 'void cgeo.geocaching.maps.interfaces.PositionAndHistory.repaintRequired()' on a null object reference
03-14 22:35:13.334 17928 17928 W cgeo    : 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4324)
03-14 22:35:13.334 17928 17928 W cgeo    : 	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4356)
03-14 22:35:13.334 17928 17928 W cgeo    : 	at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:52)
03-14 22:35:13.334 17928 17928 W cgeo    : 	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
03-14 22:35:13.334 17928 17928 W cgeo    : 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
03-14 22:35:13.334 17928 17928 W cgeo    : 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2065)
03-14 22:35:13.334 17928 17928 W cgeo    : 	at android.os.Handler.dispatchMessage(Handler.java:107)
03-14 22:35:13.334 17928 17928 W cgeo    : 	at android.os.Looper.loop(Looper.java:214)
03-14 22:35:13.334 17928 17928 W cgeo    : 	at android.app.ActivityThread.main(ActivityThread.java:7631)
03-14 22:35:13.334 17928 17928 W cgeo    : 	at java.lang.reflect.Method.invoke(Native Method)
03-14 22:35:13.334 17928 17928 W cgeo    : 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
03-14 22:35:13.334 17928 17928 W cgeo    : 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:964)
03-14 22:35:13.334 17928 17928 W cgeo    : Caused by: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=456, result=-1, data=Intent { (has extras) }} to activity {cgeo.geocaching/cgeo.geocaching.maps.google.v2.GoogleMapActivity}: java.lang.NullPointerException: Attempt to invoke interface method 'void cgeo.geocaching.maps.interfaces.PositionAndHistory.repaintRequired()' on a null object reference
03-14 22:35:13.334 17928 17928 W cgeo    : 	at android.app.ActivityThread.deliverResults(ActivityThread.java:4993)
03-14 22:35:13.334 17928 17928 W cgeo    : 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4311)
03-14 22:35:13.334 17928 17928 W cgeo    : 	... 11 more
03-14 22:35:13.334 17928 17928 W cgeo    : Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'void cgeo.geocaching.maps.interfaces.PositionAndHistory.repaintRequired()' on a null object reference
03-14 22:35:13.334 17928 17928 W cgeo    : 	at cgeo.geocaching.maps.CGeoMap.refreshMapData(CGeoMap.java:842)
03-14 22:35:13.334 17928 17928 W cgeo    : 	at cgeo.geocaching.maps.google.v2.GoogleMapActivity.onActivityResult(GoogleMapActivity.java:281)
03-14 22:35:13.334 17928 17928 W cgeo    : 	at android.app.Activity.dispatchActivityResult(Activity.java:8145)
03-14 22:35:13.334 17928 17928 W cgeo    : 	at android.app.ActivityThread.deliverResults(ActivityThread.java:4986)
03-14 22:35:13.334 17928 17928 W cgeo    : 	... 12 more
```

--- a/main/src/main/java/cgeo/geocaching/maps/CGeoMap.java
+++ b/main/src/main/java/cgeo/geocaching/maps/CGeoMap.java
@@ -850,8 +850,10 @@ public class CGeoMap extends AbstractMap implements ViewFactory, OnCacheTapListe
     @Override
     public void refreshMapData(final boolean circlesSwitched) {
         markersInvalidated = true;
-        overlayPositionAndScale.repaintRequired();
-        if (circlesSwitched) {
+        if (overlayPositionAndScale != null) {
+            overlayPositionAndScale.repaintRequired();
+        }
+        if (circlesSwitched && mapView != null) {
             mapView.setCircles(Settings.isShowCircles());
             mapView.repaintRequired(null);
         }

